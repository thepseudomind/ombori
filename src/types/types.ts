export interface IMain{
    status: boolean,
    toggleSidebar?: any
}

export interface IProduct{
    id: number,
    email: string,
    first_name: string,
    last_name: string,
    avatar: string
}

export type TLoader = {
    mini: boolean
}

export type TMessage = {
    message: string
}