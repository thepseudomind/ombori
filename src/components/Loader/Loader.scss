.loader{
    height: 7rem;
    width: 7rem;
    border-radius: 50%;
    background-color: rgba(black, .6);
    animation: pulsing 1.5s infinite cubic-bezier(0.250, 0.250, 0.750, 0.750);
    -webkit-animation: pulsing 2s infinite cubic-bezier(0.250, 0.250, 0.750, 0.750);
    -moz-animation: pulsing 2s infinite cubic-bezier(0.250, 0.250, 0.750, 0.750);
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    
    &::before{
        content: "";
        display: inline-block;
        height: 2rem;
        width: 2rem;
        border-radius: 50%;
        background-color: black;
        z-index: -1;
        position: relative;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }

    &__mini{
        height: 5rem;
        width: 5rem;
        border-radius: 50%;
        background-color: rgba(black, .6);
        animation: pulsing 2s infinite cubic-bezier(0.250, 0.250, 0.750, 0.750);
        -webkit-animation: pulsing 2s infinite cubic-bezier(0.250, 0.250, 0.750, 0.750);
        -moz-animation: pulsing 2s infinite cubic-bezier(0.250, 0.250, 0.750, 0.750);
        position: fixed;
        bottom: 2rem;
        left: 2rem; 

        &::before{
            content: "";
            display: inline-block;
            height: 1rem;
            width: 1rem;
            border-radius: 50%;
            background-color: black;
            z-index: -1;
            position: relative;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
        }
    }
}

@keyframes pulsing{
    0%{
        border: black 2rem solid;
        opacity: 1;
    }

    100%{
        border: black .5rem solid;
        opacity: .25;
    }
}