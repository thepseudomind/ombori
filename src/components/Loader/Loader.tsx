import React from 'react';
import './Loader.scss';
import { TLoader } from '../../types/types';

const Loader = ({mini}: TLoader)=>{
    return <div className={`loader${mini ? '__mini' : ''}`}></div>;
};

export default Loader;