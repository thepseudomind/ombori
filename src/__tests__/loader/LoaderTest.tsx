import React from 'react';
import { shallow } from  'enzyme';
import Loader from '../../components/Loader/Loader';
import { TLoader } from '../../types/types';

it('loader test and check if loader', ()=>{
    const mockProps : TLoader = {
        mini: false
    };
    const wrapper = shallow(<Loader {...mockProps}/>);
    expect(wrapper.find(`div.loader`).html()).toEqual('<div class="loader"></div>');
    expect(wrapper).toMatchSnapshot();
});

it('check if loader is mini', ()=>{
    const mockProps : TLoader = {
        mini: true
    };
    const wrapper = shallow(<Loader {...mockProps}/>);
    expect(wrapper.find(`div.loader__mini`).html()).toEqual('<div class="loader__mini"></div>');
});
