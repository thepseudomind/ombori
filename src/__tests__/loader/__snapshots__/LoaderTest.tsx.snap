// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader test and check if loader 1`] = `
<div
  className="loader"
/>
`;
